src/testRunner/unittests/tsbuild/javascriptProjectEmit.ts

```diff
@@ -0,0 +1,353 @@
+                "/src/common/nominal.js": utils.dedent`
+                    /**
+                     * @template T, Name
+                     * @typedef {T & {[Symbol.species]: Name}} Nominal
+                     */
+                    `,
+                "/src/common/tsconfig.json": utils.dedent`
+                    {
+                        "extends": "../../tsconfig.base.json",
+                        "compilerOptions": {
+                            "composite": true,
+                            "outFile": "common.js"
+                        },
+                        "include": ["nominal.js"]
+                    }`,
+                "/src/sub-project/index.js": utils.dedent`
+                    /**
+                     * @typedef {Nominal<string, 'MyNominal'>} MyNominal
+                     */
+                    `,
+                "/src/sub-project/tsconfig.json": utils.dedent`
+                    {
+                        "extends": "../../tsconfig.base.json",
+                        "compilerOptions": {
+                            "composite": true,
+                            "outFile": "sub-project.js"
+                        },
+                        "references": [
+                            { "path": "../common", "prepend": true }
+                        ],
+                        "include": ["./index.js"]
+                    }`,
+                "/src/sub-project-2/index.js": utils.dedent`
+                    const variable = {
+                        key: /** @type {MyNominal} */('value'),
+                    };
+
+                    /**
+                     * @return {keyof typeof variable}
+                     */
+                    function getVar() {
+                        return 'key';
+                    }
+                    `,
+                "/src/sub-project-2/tsconfig.json": utils.dedent`
+                    {
+                        "extends": "../../tsconfig.base.json",
+                        "compilerOptions": {
+                            "composite": true,
+                            "outFile": "sub-project-2.js"
+                        },
+                        "references": [
+                            { "path": "../sub-project", "prepend": true }
+                        ],
+                        "include": ["./index.js"]
+                    }`,
+                "/src/tsconfig.json": utils.dedent`
+                    {
+                        "compilerOptions": {
+                            "composite": true,
+                            "outFile": "src.js"
+                        },
+                        "references": [
+                            { "path": "./sub-project", "prepend": true },
+                            { "path": "./sub-project-2", "prepend": true }
+                        ],
+                        "include": []
+                    }`,
+                "/tsconfig.base.json": utils.dedent`

```

Comment:
I don't think this is correct sample to test unchanged d.ts. detection.. You want to change sub-project without change to its .d.ts and check output for .d.ts and .js for the final project. In current case you are changing sub-project-2 so the concatenation code isn't used.

Reply:
I think I've covered it now